Tasks/google-play-release/GooglePlay.js

```diff
@@ -227,29 +239,400 @@ function updateTrack(packageName, track, versionCode, userFraction) {
 /**
  * Add a changelog to an edit
  * Assumes authorized
- * @param {string} changeLogFile - path to changelog file. We assume this exists (behaviour may change)
- * @returns {Promise} track - A promise that will return result from updating a track
+ * @param {string} languageCode Language code (a BCP-47 language tag) of the localized listing to update
+ * @param {string} changelogFile path to changelog file. We assume this exists (behaviour may change)
+ * @returns {Promise} track A promise that will return result from updating a track
  *                            { track: string, versionCodes: [integer], userFraction: double }
  */
-function addChangelog(changeLogFile) {
-    tl.debug("Adding changelog file: " + changeLogFile);
+function addChangelog(languageCode, changelogFile) {
+    tl.debug("Adding changelog file: " + changelogFile);
+
+    var versionCode = globalParams.params.apkVersionCode;
+    try {
+        var changelogVersion = changelogFile.replace(/\.[^/.]+$/g, "");
+        versionCode = parseInt(changelogVersion);
```

Comment:
`changelogFile` is a path to the file, not the file itself. Unless I am mistaken, I don't think the regex above will pick the changelogVersion correctly.
```
var changelogFile = "c:\metadata\en-us\changelog_2.2.4\2.2.4.txt"
changelogFile.replace(/\.[^/.]+$/g, "");
"c:metadataen-uschangelog_2.2.4.2.4"
```
I would just do `path.basename(changelogFile)` which should give the filename with extension and then use the regex above. Or if you know the extension `path.basename(changelogFile, ".txt")`